### PR TITLE
[Fix] TransformedModule serialization

### PR DESCRIPTION
**Pull request by dominikandreasseitz**

### CODE REVIEW

File: tests/ml_tools/test_transformed_module.py [hunk 1]
```diff
@@ -42,7 +42,7 @@ def quantum_circuit(n_qubits: int = 2, depth: int = 1) -> QuantumCircuit:
 
 
 def get_qnn(n_qubits: int, depth: int) -> QNN:
-    observable = total_magnetization(n_qubits)
+    observable = hamiltonian_factory(n_qubits, detuning=Z)
```
Comment (Roland-djee) [inline tests/ml_tools/test_transformed_module.py:45]: Do we go fo replacing all `total_magnetization` constructors with calls to `hamiltonian_factory` ?

**Review by Roland-djee — APPROVED** [review]
LGTM.

File: tests/ml_tools/test_transformed_module.py (hunk 1 above)
Comment (dominikandreasseitz) [reply tests/ml_tools/test_transformed_module.py:45]: yes, i will do it in the MR where i clean up the tests

File: qadence/ml_tools/models.py [hunk 2]
```diff
@@ -181,7 +181,7 @@ def _transform_x(self, x: dict[str, torch.Tensor] | Tensor) -> dict[str, Tensor]
 
         else:
             assert isinstance(self.model, torch.nn.Module) and isinstance(x, Tensor)
-            return self._input_scaling * (x + self._input_shifting)
+            return self._input_scaling * x + self._input_shifting
```
Comment (nmheim) [inline qadence/ml_tools/models.py:184]: why do we want to make this change? this will break backwards compatibility...

Comment (dominikandreasseitz) [reply qadence/ml_tools/models.py]: readded!

**Pull request merged**